sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/IDigitalTwinRelationship.java

```diff
@@ -0,0 +1,69 @@
+    public static String DIGITAL_TWIN_RELATIONSHIP_NAME_JSON_PROPERTY_NAME = "$relationshipName";
+
+    /**
+     * Gets the unique Id of the relationship. This field is present on every relationship.
+     * <p>
+     * The Json property associated with this Id is {@link IDigitalTwinRelationship#DIGITAL_TWIN_RELATIONSHIP_ID_JSON_PROPERTY_NAME},
+     * so every implementation of this class must have a Json property with that value. As an example, see
+     * {@link com.azure.digitaltwins.core.models.BasicRelationship} which demonstrates how to create this
+     * property using {@link JsonProperty}.
+     *
+     * @return The unique Id of the relationship. This field is present on every relationship.
+     */
+    public String getRelationshipId();
+
+    /**
+     * Gets the unique Id of the source digital twin. This field is present on every relationship.
+     * <p>
+     * The Json property associated with this source digital twin Id is
+     * {@link IDigitalTwinRelationship#DIGITAL_TWIN_RELATIONSHIP_SOURCE_ID_JSON_PROPERTY_NAME},
+     * so every implementation of this class must have a Json property with that value. As an example, see
+     * {@link com.azure.digitaltwins.core.models.BasicRelationship} which demonstrates how to create this
+     * property using {@link JsonProperty}.
+     *
+     * @return The unique Id of the source digital twin. This field is present on every relationship.
+     */
+    public String getSourceDigitalTwinId();
+
+    /**
+     * Gets the unique Id of the target digital twin. This field is present on every relationship.
+     * <p>
+     * The Json property associated with this target digital twin Id is
+     * {@link IDigitalTwinRelationship#DIGITAL_TWIN_RELATIONSHIP_TARGET_ID_JSON_PROPERTY_NAME},
+     * so every implementation of this class must have a Json property with that value. As an example, see
+     * {@link com.azure.digitaltwins.core.models.BasicRelationship} which demonstrates how to create this
+     * property using {@link JsonProperty}.
+     *
+     * @return The unique Id of the target digital twin. This field is present on every relationship.
+     */
+    public String getTargetDigitalTwinId();
+
+    /**
+     * Gets the name of the relationship, which defines the type of link (e.g. Contains). This field is present on every relationship.
+     * <p>
+     * The Json property associated with this relationship name is
+     * {@link IDigitalTwinRelationship#DIGITAL_TWIN_RELATIONSHIP_NAME_JSON_PROPERTY_NAME},
+     * so every implementation of this class must have a Json property with that value. As an example, see
+     * {@link com.azure.digitaltwins.core.models.BasicRelationship} which demonstrates how to create this
+     * property using {@link JsonProperty}.
+     *
+     * @return The name of the relationship, which defines the type of link (e.g. Contains). This field is present on every relationship.
+     */
+    public String getRelationshipName();
```

Comment:
Same, in .NET we just call this `Name`.

Reply:
This was renamed due to the feedback above from Srikanta